detox/test/e2e/17.datePicker.test.js

```diff
@@ -1,8 +1,15 @@
 const jestExpect = require('expect').default;
 
 describe('DatePicker', () => {
+  let lastTestFailed = false;
+
   beforeEach(async () => {
-    await device.reloadReactNative();
+    if (lastTestFailed) {
```

Comment:
I'm not happy about the implied nondeterministic behavior. What about changing this to `if (android) { ...` instead?
Apropos, `lastTestFailed` never gets reset back to `false` so its going to act more like a `anyTestFailed`, ATM.

Reply:
Okay, simplified it.